### ISSUE TITLE
"cabal-fmt expand" exclusions

### ISSUE THREAD

**Issue by symbiont-sam-halliday** [phadej/cabal-fmt#4]
Thank you so much for this tool! It is going to be a regular part of my workflow.

I am able to succesfully use `-- cabal-fmt: expand <dir>` sections in my `.cabal` files when I wish to include everything. But there are cases where I do not want to include everything:

1. if I have a manually maintained `exposed-modules` section, I would like the `other-modules` section to be automatically updated and not have any duplicates.
2. if I am using `tasty-discover` and have a `Driver.hs` file as my `main-is`, then I do not want to include it in the `other-modules`.

TL;DR could `expand` of `other-modules` please automatically excludes entries from `main-is` and `exposed-modules`? I can also imagine that the user provides a manual `other-modules` and wants `exposed-modules` to be automated.

**Comment by phadej** [comment]
As I mentioned in [the blog post](https://oleg.fi/gists/posts/2019-08-11-cabal-fmt.html#s:4) putting (in your case) `exposed-modules` or `main-is` into separate folder is IMHO good enough workaround.

![image](https://user-images.githubusercontent.com/51087/62901668-0f266880-bd66-11e9-8fea-2e03897b6e1e.png)

**Comment by symbiont-sam-halliday** [comment]
![you're god damn right](https://media.giphy.com/media/3oFzmkkwfOGlzZ0gxi/giphy.gif)

**Comment by symbiont-sam-halliday** [comment]
oh, but... `tasty-discover` needs to be in the same folder as the tests. So it's not possible to use this approach there.

**Comment by symbiont-sam-halliday** [comment]
@phadej w00t! Thank you so much. Using it already.

**Comment by joprice** [comment]
I was looking to remove hpack, since I only use it for other-module generation and I need less yaml in my life. Using pragmas and multiple folders works great. The only thing I didn't realize at first is that I needed `hs-source-dirs` to include the public and private folders, otherwise you get errors like `cabal: can't find source for Server in src`. I'm relatively new to haskell and assumed that the folder structure and module names had to match, which would force someone to name the module `Public.X` / `Private.X`. This pragma a great feature and should be migrated to cabal!
